code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm

```diff
@@ -61,6 +62,26 @@
 	bolt_act(user)
 
 /obj/item/gun/projectile/boltgun/proc/bolt_act(mob/living/user)
+
+	if(bolting)
+		return FALSE
+	
+	bolting = TRUE
+	var/datum/progressbar/progbar
+	var/delay = 0.3 SECONDS
+	progbar = new(user, delay, src)
+
+	var/endtime = world.time + delay
+	var/starttime = world.time
+	. = 1
+	while (world.time < endtime)
+		sleep(1)
+		progbar.update(world.time - starttime)
+
+	if (progbar)
+		qdel(progbar)
```

Comment:
add a mobile flag to do_after instead of hard-coding this,  its usefull in general , if you need a PR to rip it off from , try the wielding one.